File: framework/java/src/main/java/soda/web/SetupHandler.java
package soda.web;

import com.sun.net.httpserver.HttpExchange;
import soda.web.http.RequestHelper;

public class SetupHandler extends BaseHandler {
	
	private final ClassLoaderManager mgr;
	
	public SetupHandler(ClassLoaderManager mgr) {
		this.mgr = mgr;
	}

	@Override
	protected void doPost(HttpExchange exchange) throws Exception {
		// multipart/form-data, key = ?, jar = ?
		var data = RequestHelper.multipartFormData(exchange);
		var key = data.firstValue("key");
		mgr.remove(key);

		var bytes = data.firstFile("jar");
		mgr.setupForJar(key, bytes);

		Logger.infof("reset class loader for key %s", key);

		Utils.setResponse(exchange, 200, "success");
	}

}

File: framework/scala/src/main/scala/soda/scala/web/SetupHandler.scala
package soda.scala.web
import com.sun.net.httpserver.HttpExchange

class SetupHandler extends BaseHandler {
  override def handleWork(exchange: HttpExchange): String = {
    // multipart/form-data, key = ?, jar = ?
    val parts = parseMultipart(exchange)
    val key: String = parts("key").bodyString
    ClassLoaderCache.remove(key)

    val bytes: Array[Byte] = parts("jar").payload
    ClassLoaderCache.setupForJar(key, bytes)

    s"reset class loader with $key"
  }
}
